src/App.tsx

```diff
@@ -1,24 +1,147 @@
+    setWaitingResponse([id]);
+    deleteTodo(id)
+      .then(() => {
+        setTodos(todos.filter((todo) => todo.id !== id));
+        setWaitingResponse([]);
+      })
+      .catch(() => {
+        setErrors({ deleting: true });
+      });
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((todosList) => {
+        setTodos(todosList);
+      })
+      .catch(() => {
+        setErrors({ loading: true });
+      });
+  }, []);
+
+  const filterTodos = todos.filter((todo) => {
+    switch (filter) {
+      case Filter.All:
+        return todo;
+
+      case Filter.Active:
+        return !todo.completed;
+
+      case Filter.Completed:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          addTodo={addTodo}
+          todos={todos}
+          setTodos={setTodos}
+          setErrors={setErrors}
+          tempTodo={tempTodo}
+          setWaitingResponse={setWaitingResponse}
+        />
+
+        {todos.length > 0 && (
+          <Main
+            filteredTodos={filterTodos}
+            tempTodo={tempTodo}
+            removeTodo={removeTodo}
+            waitingResponse={waitingRespone}
+            setWaitingResponse={setWaitingResponse}
+            todos={todos}
+            setTodos={setTodos}
+            setErrors={setErrors}
+          />
+        )}
+
+        {/* Hide the footer if there are no todos */}
+        {todos.length > 0 && (
+          <Footer
+            filter={filter}
+            filteredTodos={filterTodos}
+            setTodos={setTodos}
+            setFilter={setFilter}
+            todos={todos}
+            setErrors={setErrors}
+            setWaitingResponse={setWaitingResponse}
+          />
+        )}
+      </div>
+
+      {/* Notification is shown in case of any error */}
+      {/* Add the 'hidden' class to hide the message smoothly */}
+      {errors && (
+        <div className="notification is-danger is-light has-text-weight-normal">
+          <button
+            type="button"
+            onClick={() => setErrors(null)}
+            className="delete"
+          />
+          {/* show only one message at a time */}
+          {errors?.noTitle && 'Title cannot be empty'}
+          {errors?.loading && 'Unable to load todos'}
+          {errors?.posting && 'Unable to add a todo'}
+          {errors?.editing && 'Unable to update a todo'}
+          {errors?.deleting && 'Unable to delete a todo'}
```

Comment:
Make a 1 state with error
```suggestion
          {error}
```